adapter/pkg/api/file_read_utils.go

```diff
@@ -44,35 +57,34 @@ func ApplyAPIProject(payload []byte) error {
 
 	// TODO: (VirajSalaka) this won't support for distributed openAPI definition
 	for _, file := range zipReader.File {
-		if strings.HasSuffix(file.Name, "Meta-information/swagger.yaml") {
+		if strings.HasSuffix(file.Name, openAPIDir+string(os.PathSeparator)+openAPIFilename) {
 			loggers.LoggerAPI.Debugf("openAPI file : %v", file.Name)
 			unzippedFileBytes, err := readZipFile(file)
 			if err != nil {
 				loggers.LoggerAPI.Errorf("Error occured while reading the openapi file. %v", err.Error())
 				continue
 			}
-			apiJsn, conversionErr := utills.ToJSON(unzippedFileBytes)
+			apiJsn, conversionErr = utills.ToJSON(unzippedFileBytes)
 			if conversionErr != nil {
 				loggers.LoggerAPI.Errorf("Error converting api file to json: %v", err.Error())
 				return conversionErr
 			}
-			xds.UpdateEnvoy(apiJsn)
+		} else if strings.Contains(file.Name, endpointCertDir+string(os.PathSeparator)) &&
+			(strings.HasSuffix(file.Name, crtExtension) || strings.HasSuffix(file.Name, pemExtension)) {
+			//TODO: (VirajSalaka) Validate the content of cert files
```

Comment:
Validate the content with a regex

Reply:
fixed via https://github.com/wso2/product-microgateway/pull/1525/commits/342713069cbd72aa063300e9e3220425dc4262a5

Reply:
Please note that this check is just based on surrounding "----BEGIN CERTIFICATE----" and "-----END CERTIFICATE---" strings.